app/Models/Score/Model.php

```diff
@@ -54,6 +54,18 @@ public static function getMode(): string
         return snake_case(get_class_basename(static::class));
     }
 
+    public function difficultyRating(): ?float
```

Comment:
If this is to actually exist here, it at very least needs a better name. A score can't have a "difficulty rating", so it would need a beatmap prefix.

Optimally I don't think it should be in here at all, but will defer to the web team to ascertain whether a structure to make that happen is possible.

Reply:
the closest thing I can think of is making it a relation called `beatmapDifficulty` instead.

Similarly for the transformer although considering difficulty rating is the only relevant value `beatmap_difficulty_rating` could be sufficient...

Reply:
looking again, relation is quite difficult because the difficulty table uses composite primary key. Short of adding virtual column and indexing it, it's probably better being an extra select column with subquery similar to [this](https://github.com/ppy/osu-web/blob/e50b1a4a16efc11fd3f491c332222a1326ac5788/app/Models/Multiplayer/Room.php#L219-L244) (although with simpler query).

Reply:
And if adding skipping difficulty query, then both mode and mods should be checked in case of convert maps (which would have different difficulty rating even when no other mods).

Oh and don't bother going through Beatmap model.